README.md

```diff
@@ -39,98 +41,165 @@ Current stable release is able to persist Orion context data in:
+1.  On behalf of Cygnus, subscribe to certain NGSI-like source (typically Orion Context Broker) for certain context information.
+2.  Receive from the NGSI-like source notifications about new updated context data; this notification will be handled by the native `HttpSource` together with the custom `OrionRestHandler`.
+3.  Translate the notification into the Flume event format (metadata headers + data body), and put them into the different sink channels, typically of type `MemoryChannel`.
+4.  In the meantime, some interceptors such as the native `Timestamp` one or the custom `DestinationExtractorInterceptor` may modify the event before it is put in the channel or channels.
+5.  For each enabled custom sink (`OrionHDFSSink`, `OrionCKANSink`, `OrionMySQLSink`, `OrionMongoSink`), get the Flume events from the sink channels and persist the data in the appropriate format.
+
+More complex architectures and data flows can be checked in the [architecture](doc/design/architecture.md) document.
 
 [Top](#top)
 
-##<a name="section3"></a>Functionality explained (Json notification example)
+##<a name="section3"></a>Data flow example
+Next sections will consider an example NGSI entity called 'car1' of type 'car', with attributes 'speed' (type 'kmh') and 'oil_level' (type 'percentage'). Is not a goal for this document to show you how to define a NGSI entity nor how to create it in the most common NGSI source for Cygnus, Orion Context Broker. Please, refer to the [official Orion documentation](https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Publish/Subscribe_Broker_-_Orion_Context_Broker_-_User_and_Programmers_Guide#Entity_Creation) for more details.
 
-Let's consider the following notification in Json format coming from an Orion Context Broker instance:
+[Top](#top)
 
-    POST http://localhost:1028/notify
-    Content-Length: 492
-    User-Agent: orion/0.9.0
-    Host: localhost:1028
-    Accept: application/xml, application/json
-    Content-Type: application/json
-    Fiware-Service: my-company-name
-    Fiware-ServicePath: /workingrooms/floor4 
-    
+###<a name="section3.1"></a>Subscription to the NGSI-like source
+Cygnus takes advantage of the subscription-notification mechanism of NGSI. Specifically, Cygnus needs to be notified each time certain entity's attributes change, and in order to do that, Cygnus must subscribe to those entity's attribute changes.
+
+As long as the typical NGSI-like source is Orion Context Broker, you can make a subscription about the example NGSI entity ('car1' of type 'car') by using the `curl` command in this [way](https://forge.fi-ware.eu/plugins/mediawiki/wiki/fiware/index.php/Publish/Subscribe_Broker_-_Orion_Context_Broker_-_User_and_Programmers_Guide#ONCHANGE) (assuming Orion runs in localhost and listens on the TCP/1026 port):
+
+    (curl localhost:1026/v1/subscribeContext -s -S --header 'Content-Type: application/json' --header 'Accept: application/json' -d @- | python -mjson.tool) <<EOF
```

Comment:
Since already linked to the Context Broker, you could use this links as suggested by @fgalan :
  - Service: https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Publish/Subscribe_Broker_-_Orion_Context_Broker_-_User_and_Programmers_Guide#Multi_service_tenancy
  - Service path: https://forge.fiware.org/plugins/mediawiki/wiki/fiware/index.php/Publish/Subscribe_Broker_-_Orion_Context_Broker_-_User_and_Programmers_Guide#Entity_service_paths